torch/nn/modules/module.py

```diff
@@ -1668,6 +1672,16 @@ def _register_state_dict_hook(self, hook):
         self._state_dict_hooks[handle.id] = hook
         return handle
 
+    def register_state_dict_pre_hook(self, hook):
+        r"""These hooks will be called with arguments: ``self``, ``prefix``,
+        and ``keep_vars`` before calling ``state_dict`` on ``self``. The registered
```

Comment:
A comment from the post hook discussion is that since `state_dict()` is not actually called recursively, this hook will only be triggered on the top level Module.
We should clarify that in the doc if that is expected. Or think about updating the implementation/testing if it is not.